### PR TITLE
Fixes Unit Map Testing selection

### PR DESCRIPTION
**Pull request by CPU-Blanc**
Fixes the ability to actually select which map is tested during the unit tests.
Due to the away maps and example map failing the checks, this'll be a draft PR until I fix those to comply.

Todo:

- [x] Fix test failures on `away_map_testing`
- [x] Fix test failures on `example`
- ~~Fix template testing to not cause OOM errors~~
- [x] Re-test once #1169  is merged

### CODE REVIEW

**Comment by CPU-Blanc** [comment]
All unit tests now passing!

Contains a fix to the bucket timer, originally from: https://github.com/tgstation/tgstation/pull/72928

Blocked until #1169 is merged to check for any new issues from the new unit tests introduced there

**Comment by Glloyd** [comment]
You have conflicts from #1172

**Comment by CPU-Blanc** [comment]
Merge conflicts resolved, and all test fails introduced with the new tests from #1169 and some errors from #1172 have been fixed.

@Glloyd if you want to get this in for #1165 , I can reduce the scope and drop the template testing fix? I can open that in another PR and then this one will be ready for review?

**Comment by CPU-Blanc** [comment]
Yep. Shouldn't be any issues with existing tests. Current plan is to add another workflow test that sest a new flag in the `test.sh` script that'll toggle template testing, and do it on a blank map to maximise memory. Got it to work already, just have errors to fix.

I'll open that in a later PR. This one is ready for review + merge.

File: code/__defines/urist/maths.dm
Comment (Glloyd) [inline code/__defines/urist/maths.dm:1]: unrelated to this pr but we should probably move the things in uristdefines.dm up to this folder.

File: baystation12.dme
Comment (Glloyd) [inline baystation12.dme:3639]: why was this moved to the .dme? It might cause issues loading glloydstation.

**Review by Glloyd — CHANGES_REQUESTED** [review]

File: baystation12.dme
Comment (CPU-Blanc) [reply baystation12.dme:3639]: You told me to in the discord :P 
It was moved because it was required in `away_site_testing`, but this was only ever included in the Nerva code. If it's in the .dme, all maps will have access to it, so it shouldn't cause issues for glloydstation?

**Comment by CPU-Blanc** [comment]
There was one compile error I accidentally created with this when compiling Glloydstation. Whoops.

I quickly ran the tests over Glloydstation, and while it does run, there's many failures. Nothing relating to any of the new test PRs, but must've been tests expanded in the baymerge. These will need to be looked at sometime (yaaay guess who)

File: baystation12.dme
Comment (Glloyd) [reply baystation12.dme:3639]: i'm referring to glloyd_jungle.dm. The geminus stuff is fine to be in the dme, but that shouldn't because it's the actual awaymap definition. The jungle on glloydstation is just part of the map, not an away.

Comment (Glloyd) [reply baystation12.dme:3639]: On that thread though, there are a couple nerva awaymaps missing from away_site_testing it seems

Comment (CPU-Blanc) [reply baystation12.dme:3639]: Ah. You know, I don't remember why. I'll re-run some tests with it back and get back to you

Comment (CPU-Blanc) [reply baystation12.dme:3639]: Re: Missing awaymaps, as said on a different PR, I'm tempted to write a mini test that'll scream if an awaymap isn't included in `away_sites_testing.dm`. I assume basically all aways should be in there?

Comment (Glloyd) [reply baystation12.dme:3639]: yeah, i can't think of any reason why one shouldn't be in there as long as it's in use.

Comment (CPU-Blanc) [reply baystation12.dme]: No idea why I moved the jungle. My bad. Moved it back to Nerva & added it to away testing

**Review by Glloyd — APPROVED** [review]
Good stuff 👍 

Thanks for this Blanc

**Pull request merged**
